tests/metrics/metric_manager/test_metric_manager.py

```diff
@@ -0,0 +1,152 @@
+import unittest
+
+from src.constructor import METRICS
+from src.metrics.metric_manager import MetricParams, MetricManager, Phase
+
+import torch
+from torchmetrics import Metric
+from typing import List, Dict
+
+from .data_generator import FakeData, FakeDataGenerator
+
+
+@METRICS.register_class
+class MocSumMetric(Metric):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.add_state('lol_sum', default=torch.tensor(0), dist_reduce_fx=None)
+
+    def update(self, predict: torch.Tensor, target: torch.Tensor):
+        self.lol_sum += 1
+
+    def compute(self):
+        return self.lol_sum
+
+
+@METRICS.register_class
+class MockSumMetric(Metric):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.add_state('sum', default=torch.tensor(0), dist_reduce_fx=None)
+
+    def update(self, predict: torch.Tensor, target: torch.Tensor):
+        self.sum += 1
+
+    def compute(self):
+        return self.sum
+
+
+@METRICS.register_class
+class MockConstantMetric(Metric):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.add_state('constant', default=torch.tensor(0), dist_reduce_fx=None)
+
+    def update(self, predict: torch.Tensor, target: torch.Tensor):
+        return
+
+    def compute(self):
+        return self.constant
+
+
+@METRICS.register_class
+class MocRaiseMetric(Metric):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def update(self, predict: torch.Tensor, target: torch.Tensor):
+        return
+
+    def compute(self, memory_block):
+        return torch.tensor([1, 2])
+
+
+def run_metric_manager(class_names: List[str], names: List[str], \
```

Comment:
Maybe to call arguments as they should be called? `metric_names`, `log_names`, `mappings`